pkg/collector/corechecks/servicediscovery/module/stat.go

```diff
@@ -47,3 +50,78 @@ func getRSS(proc *process.Process) (uint64, error) {
 
 	return rssPages * pageSize, nil
 }
+
+func getGlobalCPUTime() (uint64, error) {
+	globalStatPath := kernel.HostProc("stat")
+
+	// This file is very small so just read it fully.
+	file, err := os.Open(globalStatPath)
+	if err != nil {
+		return 0, err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	// Try to read the first line; it contains all the info we need.
+	if !scanner.Scan() {
+		return 0, scanner.Err()
+	}
+
+	// See proc(5) for a description of the format of statm and the fields.
+	fields := strings.Fields(scanner.Text())
+	if fields[0] != "cpu" {
+		return 0, errors.New("invalid /proc/stat file")
+	}
+
+	var totalTime uint64
+	for _, field := range fields[1:] {
+		val, err := strconv.ParseUint(field, 10, 64)
+		if err != nil {
+			return 0, err
+		}
+		totalTime += val
+	}
+
+	return totalTime, nil
+}
+
+func updateCPUUsageStats(proc *process.Process, info *serviceInfo, lastGlobalCPUTime, currentGlobalCPUTime uint64) (float64, error) {
+	statPath := kernel.HostProc(strconv.Itoa(int(proc.Pid)), "stat")
+
+	// This file is very small so just read it fully.
+	content, err := os.ReadFile(statPath)
+	if err != nil {
+		return 0, err
+	}
+
+	startIndex := bytes.LastIndexByte(content, byte(')'))
+	if startIndex == -1 || startIndex+1 >= len(content) {
+		return 0, errors.New("invalid stat format")
+	}
+
+	// See proc(5) for a description of the format of statm and the fields.
+	fields := strings.Fields(string(content[startIndex+1:]))
+	if len(fields) < 50 {
+		return 0, errors.New("invalid stat format")
+	}
+
+	// Parse fields at index 15 and 16, resp. User and System CPU time.
```

Comment:
Maybe expand this comment to say why you're addressing index 11 and 12 when the comment says 15 and 16?